src/components/ProductForm/ProductForm.js

```diff
@@ -0,0 +1,79 @@
+import React, { useState } from 'react';
+import { Link } from 'react-router-dom';
+const ProductForm = ({
+  name,
+  id,
+  image,
+  price,
+  rate,
+  disPrice,
+  gender,
+  status,
+}) => {
+  const [isHeart, setIsHeart] = useState(true);
+
+  const heartColorChange = () => {
+    setIsHeart(!isHeart);
+  };
+  return (
+    <li className="productForm">
+      <div className="thumb">
+        <span className={`num ${!(id < 5) && 'new'}`}>
+          {id >= 1 && id < 9 && `${id}`}
+        </span>
+        <Link
+          key={id}
+          to={`/detailproduct/${id}`}
+          state={{ gender: gender, status: status }}
+          className="images"
+        >
+          <div className="list">
+            <img
+              onClick={() => console.log('zzz')}
+              className="menu"
+              src="/images/menu.png"
+              alt="menu"
+            />
+          </div>
+          <img src={image} alt="제품 사진" />
+        </Link>
+        <div className="description">
+          <div className="name">
+            <span>{name}</span>
+            {isHeart ? (
+              <img
+                onClick={heartColorChange}
+                className="heart"
+                src="/images/heart.png"
+                alt="heart"
+              />
+            ) : (
+              <img
+                onClick={heartColorChange}
+                className="heart"
+                src="/images/heartPink.png"
+                alt="heart"
+              />
+            )}
+          </div>
+          {rate ? (
+            <div className="price">
+              {Math.floor(disPrice).toLocaleString()}
+              <span className="originalPrice">
+                {Math.floor(price).toLocaleString()}
+              </span>
+              <span className="rateDiscount">{`${rate}%`}</span>
+            </div>
+          ) : (
+            <div className="price">{Math.floor(disPrice).toLocaleString()}</div>
+          )}
```

Comment:
```suggestion
  const floorPrice = (price) => {
    return Math.floor(price).toLocaleString()
  }

          {rate ? (
            <div className="price">
              {floorPrice(disPrice)}
              <span className="originalPrice">
                {floorPrice(price)}
              </span>
              <span className="rateDiscount">{`${rate}%`}</span>
            </div>
          ) : (
            <div className="price">{floorPrice(disPrice)}</div>
          )}
```
이렇게 반복되는 부분은 함수 혹은 변수를 활용해 보세요!

Reply:
감사합니다 도현님 수정  완료 헀습니다.